### PR TITLE
[Tags Feed] Fix duplicate tracking in Reader subfilter

### LINKED ISSUE #20814: [Reader] Fix `reader_filter_sheet_item_selected` tracked multiple times
`reader_filter_sheet_item_selected` should be tracked once when the selected filter changes, but it's being tracked multiple times.

### PR DESCRIPTION
**Pull request by thomashorta**
Fixes #20814 

- Keep a consistent reference to the SubFilterViewModel observers to leverage the existing Observable APIs, which handle duplicate observation, removal, and re-observation in case of multiple observe call and lifecycle state changes
- Fix sending the `reader_filter_sheet_item_selected` when selecting a filterable feed (e.g.: Subscriptions and Your Tags)

-----

## To Test:

1. Install Jetpack
2. Make sure `App Setting` -> `Privacy` -> `Collect Information` is enabled
3. Make sure `reader_tags_feed` Feature config is enabled (`Debug Settings` -> `Remote Features`
4. Go to Reader
5. Go to Subscriptions
6. **Verify** `reader_filter_sheet_item_selected` is not tracked
7. Tap the subfilter pill
8. **Verify** appropriate events are tracked ONLY ONCE
9. Tap a Site
10. **Verify** `reader_filter_sheet_item_selected` is tracked ONLY ONCE
11. Tap the pill again
12. Repeat 9-11 multiple times making sure events are tracked ONLY ONCE per trigger
13. Go to Your Tags
14. Repeat 6-12

-----

## Regression Notes

1. Potential unintended areas of impact

    - N/A

15. What I did to test those areas of impact (or what existing automated tests I relied on)

    - N/A

16. What automated tests I added (or what prevented me from doing so)

    - Unit tests not added for the one-time initialization mechanism since it is done completely in Fragment and ViewModel non-exposed APIs (`onCleared`)
    - Unit tests updated for bug fix related to sending `reader_filter_sheet_item_selected` when opening a filterable feed

-----

## PR Submission Checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

-----

## Testing Checklist (strike-out the not-applying and unnecessary ones):

- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [x] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] Talkback.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] Large and small screen sizes. (Tablet and smaller phones)
- [ ] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

### CODE REVIEW

**Review by daniloercoli — APPROVED** [review]
LGTM! 
While testing this PR i've found a bug in the top menu. We will discuss it on Slack before opening a new ticket for it.

**Comment by thomashorta** [comment]
I tested the scenarios we discussed on Slack and indeed the solution here doesn't seem to be working correctly for some of them. I'm marking the PR as `Not Ready for Merge` and `Not Ready for Review` while I think of a different solution. Thanks for the review @daniloercoli !

**Comment by thomashorta** [comment]
@daniloercoli, to fix the issue I needed to radically change the solution. The new solution properly uses the observable mechanism by creating a local reference to the Observers, which prevents duplicate observation by default (thanks to the "Observer, LifecycleOwner" tuple being stable and consistent now), and also handles lifecycle and view model state changes automatically.

Please review again with the new changes!

File: WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderFragment.kt [hunk 1]
```diff
@@ -96,6 +97,86 @@ class ReaderFragment : Fragment(R.layout.reader_fragment_layout), ScrollableView
...
 
+    private val wpMainActivityViewModel by lazy {
+        ViewModelProvider(
+            requireActivity(),
+            viewModelFactory
+        )[WPMainActivityViewModel::class.java]
+    }
+
+    // region SubgroupFilterViewModel Observers
+    // we need a reference to the observers so they are properly handled by the lifecycle and ViewModel owners, avoiding
```
Comment (daniloercoli) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderFragment.kt:108]: 👍

**Comment by daniloercoli** [comment]
> which prevents duplicate observation by default (thanks to the "Observer, LifecycleOwner" tuple being stable and consistent now), and also handles lifecycle and view model state changes automatically.

Tested and cannot replicate the issue anymore! Code looks robust, and adheres to good practices with regard to ViewModel and lifecycle management. LGTM!

**Review by daniloercoli — APPROVED** [review]

**Comment by daniloercoli** [comment]
Feel free to merge this PR, once the problems with testes are resolved.

**Pull request merged**